eth_account/signers/ledger.py

```diff
@@ -0,0 +1,399 @@
+                if ('interface_number' in hidDevice and hidDevice['interface_number'] == 0) or ('usage_page' in hidDevice and hidDevice['usage_page'] == 0xffa0):
+                    hidDevicePath = hidDevice['path']
+        if hidDevicePath is not None:
+            dev = hid.device()
+            dev.open_path(hidDevicePath)
+            dev.set_nonblocking(True)
+        else:
+            raise Exception("No dongle found")
+        self.device = dev
+        self.debug = debug
+
+    def exchange(self, apdu, timeout=20000):
+        if self.debug:
+            print("HID => %s" % to_hex(apdu))
+
+        # Construct the wraped payload
+        apdu = self.wrapCommandAPDU(0x0101, apdu, 64)
+        padSize = len(apdu) % 64
+        tmp = apdu
+        if padSize != 0:
+            tmp.extend([0] * (64 - padSize))
+
+        # Send to device
+        offset = 0
+        while(offset != len(tmp)):
+            data = tmp[offset:offset + 64]
+            data = bytearray([0]) + data
+            if self.device.write(data) < 0:
+                raise BaseException("Error while writing")
+            offset += 64
+
+        # Receive reply
+        dataLength = 0
+        dataStart = 2
+        result = self.waitFirstResponse(timeout)
+        self.device.set_nonblocking(False)
+        while True:
+            response = self.unwrapResponseAPDU(0x0101, result, 64)
+            if response is not None:
+                result = response
+                dataStart = 0
+                swOffset = len(response) - 2
+                dataLength = len(response) - 2
+                self.device.set_nonblocking(True)
+                break
+            result.extend(bytearray(self.device.read(65)))
+        sw = (result[swOffset] << 8) + result[swOffset + 1]
+        response = result[dataStart : dataLength + dataStart]
+        if self.debug:
+            print("HID <= %s  %.2x" % (to_hex(response), sw))
+        if sw != 0x9000:
+            possibleCause = "Unknown reason"
+            if sw == 0x6982:
+                possibleCause = "Have you uninstalled the existing CA with resetCustomCA first?"
+            if sw == 0x6985:
+                possibleCause = "Condition of use not satisfied (denied by the user?)"
+            if sw == 0x6a84 or sw == 0x6a85:
+                possibleCause = "Not enough space?"
+            if sw == 0x6484:
+                possibleCause = "Are you using the correct targetId?"
+            raise Exception("Invalid status %04x (%s)" % (sw, possibleCause))
+        return response
+
+    def waitFirstResponse(self, timeout):
+        start = time.time()
+        data = ""
+        while len(data) == 0:
+            data = self.device.read(65)
+            if not len(data):
+                if time.time() - start > timeout:
+                    raise Exception("Timeout")
+                time.sleep(0.0001)
+        return bytearray(data)
+
+    def wrapCommandAPDU(self, channel, command, packetSize):
```

Comment:
This method doesn't appear to use `self`.  What do you think about extracting it to a utility function so that it can have some basic tests written for it (which will also allow it to be easily tested in isolation)